blocks/library/tag-cloud/block.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+import { stringify } from 'querystringify';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import {
+	Placeholder,
+	ToggleControl,
+	SelectControl,
+	withAPIData,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TermList from './term-list';
+import InspectorControls from '../../inspector-controls';
+
+class TagCloudBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.setTaxonomy = this.setTaxonomy.bind( this );
+		this.toggleShowTagCounts = this.toggleShowTagCounts.bind( this );
+	}
+
+	getTaxonomies() {
+		const taxonomies = this.props.taxonomies.data;
+
+		return map( taxonomies, taxonomy => {
+			return {
+				value: taxonomy.slug,
+				label: taxonomy.name,
+			};
+		} );
+	}
+
+	setTaxonomy( taxonomy ) {
+		const { setAttributes } = this.props;
+
+		setAttributes( { taxonomy } );
+	}
+
+	toggleShowTagCounts() {
+		const { attributes, setAttributes } = this.props;
+		const { showTagCounts } = attributes;
+
+		setAttributes( { showTagCounts: ! showTagCounts } );
+	}
+
+	render() {
+		const {
+			attributes,
+			focus,
+			termList,
+			className,
+		} = this.props;
+		const { taxonomy, showTagCounts } = attributes;
+		const taxonomies = this.getTaxonomies();
+		const options = [
+			{
+				label: __( '- Select -' ),
+				value: '',
+			},
+			...taxonomies,
+		];
+		const terms = termList.data ? (
+			<TermList
```

Comment:
We could avoid having to implement `TermList` which is a duplication of `wp_tag_cloud()` if we used `ServerSideRender` proposed in #780.

Reply:
I agree, not sure what the status of that proposal is but it would have
been useful.

Reply:
@jahvi the `ServerSideRender`component has landed with https://github.com/WordPress/gutenberg/pull/5495

Do you want to refactor your PR to make use of it?

Reply:
Ah that's interesting, I'll refactor my PR then

Reply:
Just updated the code to use the new `ServerSideRender` component, it does simplify things quite a lot 🙏🏽